lib/std/json.zig

```diff
@@ -1586,7 +1586,11 @@ fn parseInternal(comptime T: type, token: Token, tokens: *TokenStream, options:
                                         return error.UnexpectedValue;
                                     }
                                 } else {
-                                    @field(r, field.name) = try parse(field.field_type, tokens, options);
+                                    if (fields_seen[i]) {
```

Comment:
I'd add a check here for `options.duplicate_field_behavior == .UseFirst` here that would make the branch predictor happy.

I also was hoping that we could eventually make `duplicate_field_behavior` comptime-known.